hazelcast/src/main/java/com/hazelcast/internal/partition/impl/PartitionReplicaManager.java

```diff
@@ -352,17 +371,55 @@ void cancelReplicaSync(int partitionId) {
             if (syncInfo.partitionId == partitionId) {
                 iter.remove();
                 replicaSyncTimeoutScheduler.cancel(syncInfo);
-                releaseReplicaSyncPermit();
+                releaseReplicaSyncPermits(1);
             }
         }
     }
 
-    public boolean tryToAcquireReplicaSyncPermit() {
-        return replicaSyncProcessLock.tryAcquire();
+    /**
+     * Tries to acquire requested permits. Less than requested permits may be acquired,
+     * if insufficient permits are available. Number of actually acquired permits will be
+     * returned to the caller. Acquired permits will be in the range of {@code [0, requestedPermits]}.
+     *
+     * @param requestedPermits number of requested permits
+     * @return number of actually acquired permits.
+     */
+    public int tryAcquireReplicaSyncPermits(int requestedPermits) {
+        assert requestedPermits > 0 : "Invalid permits: " + requestedPermits;
+
+        int permits = Math.min(requestedPermits, availableReplicaSyncPermits());
```

Comment:
Yes, right. My purpose was to avoid extra iterations in the try-acquire loop. 
But actually acquiring less than maximum possible permits will cause more distributed `request->retry` cycles, which is worse than a few more local loop iterations.  
I'll remove this in master port.